tutorial-murrayobrien/blog.md

```diff
@@ -0,0 +1,199 @@
+            console.error(error);
+         });
+      return word;
+   };
+
+   //STEP 3: We'll need to break this function down further but I'll try explain each line with comments.
+   export const sendWordOfTheDay = async (wordOfTheDay: IWordOfTheDay) => {
+      //3.1 Fetch users, I used Firebase to store and fetch my users but I'll leave that up to you, so for now we'll hardcode our users.
+      const users = [
+         email: "abc@gmail.com", language: "es",
+         email: "test@gmail.com", language: "ge",
+      ];
+      //Using npm package lodash, I'm combining users with same chosen language into a single array.
+      const groupedByLanguage: any = _.values(_.groupBy(users, "language"));
+      //translate word for all the user
+      for (const users of groupedByLanguage) {
+         //get language of grouped users
+         const language = users[0].language;
+         //Translate our word:
+         const translatedWord = await translateWord(
+            wordOfTheDay.translation,
+            language
+         );
+         let newWordOfTheDay = wordOfTheDay;
+         newWordOfTheDay.word = translatedWord;
+         //send to users who subscribe to the same language
+         await mailWord(users, newWordOfTheDay);
+      }
+   };
+```
+
+### Part 3: Send our users the translated word and its description
+
+Finally, let's have a look at our method for sending the notification to our users. This is done in our function called mailWord() which is called in step 3 above this. This function requires an array of users who are subscribed to the same language and the WordOfTheDay object which contains the translated word and its description. This function also uses an email template that I created using Couriers Notification Designer where you can pass variables. Your template can be used and referenced below by an id which you can find on the Notification Designer.
+
+```typescript
+export const mailWord = async (users: IUser[], wordOfTheDay: IWordOfTheDay) => {
+  //Initialize courier client from secret variable specified in your .env file
+  const courier = CourierClient({
+    authorizationToken: process.env.courier_api_key,
+  });
+
+  //Create an array of objects so that Courier can send to all these users with one API call.
+  const list = users.map((user) => {
+    return {
+      email: user.email,
+      data: {
+        name: user.email,
+        word: wordOfTheDay.word,
+        language: getLanguageNameByCode(user.language),
+        translation: wordOfTheDay.translation,
+        description: wordOfTheDay.description,
+      },
+    };
+  });
+
+  //Final step is send.
+  const { requestId } = await courier.send({
+    message: {
+      to: list,
+      template: "HECEE75KFB4V01PTKX7F06H6F5ZP",
+    },
+  });
+};
+```
+
+![Image of Template](./images/template.png)
+
+## Conclusions
+
+Go check the project out over at Devpost : https://devpost.com/software/word-of-the-day-4l6rqj
```

Comment:
What are some challenges you faced while building this? How would you suggest the readers challenge themselves to take this project to the next level? What was your experience like using Courier?